### ISSUE TITLE
Cannot filter using has_id, HasIdCondition model is not serialized correctly

### ISSUE THREAD

**Issue by illagrenan** [qdrant/qdrant_client#11]
According to https://qdrant.github.io/qdrant/redoc/index.html#operation/search_points, the `filter` field should be one of `FieldCondition` or `HasIdCondition` or `Filter` (recursively).

I'm unable to use `HasIdCondition` (other filters work fine). `HasIdCondition` does not seem to serialize correctly.

Example code:

```python
from qdrant_openapi_client.models.models import (
    FieldCondition,
    Filter,
    HasIdCondition,
    Match,
)

Filter(
    must=[
        HasIdCondition(has_id=[42, 43]),
        FieldCondition(key="field_name", match=Match(keyword="field_value_42")),
    ]
).json()
```

Result:

```
{
  "must": [
    {
      "must": null,
      "must_not": null,
      "should": null
    },
    {
      "geo_bounding_box": null,
      "geo_radius": null,
      "key": "field_name",
      "match": {
        "integer": null,
        "keyword": "field_value_42"
      },
      "range": null
    }
  ],
  "must_not": null,
  "should": null
}
```

Note that the `has_id` is completely missing.

I would expect an output like this:

```json
{
  "must": [
    {
      "has_id": [
        42,
        43
      ]
    },
    {
      "geo_bounding_box": null,
      "geo_radius": null,
      "key": "field_name",
      "match": {
        "integer": null,
        "keyword": "field_value_42"
      },
      "range": null
    }
  ],
  "must_not": null,
  "should": null
}
```

I use `qdrant-client==0.3.11` and Python `3.9`. I tried upgrading Pydantic, but the output was the same.

**Comment by illagrenan** [comment]
I have found that this is documented Pydantic behavior. The problem is the order of `qdrant_openapi_client.models.models.Condition`.

> As such, it is recommended that, when defining Union annotations, the most specific type is included first and followed by less specific types. In the above example, the UUID class should precede the int and str classes to preclude the unexpected representation as such:

Ref.: https://pydantic-docs.helpmanual.io/usage/types/#unions

There is the following workaround:

```python
from typing import Union

from qdrant_openapi_client.models import models
from qdrant_openapi_client.models.models import (
    FieldCondition,
    Filter,
    HasIdCondition,
    Match,
)

models.Condition = Union[
    FieldCondition,
    HasIdCondition,
    Filter,
]

Filter(
    must=[
        HasIdCondition(has_id=[42, 43]),
        FieldCondition(key="field_name", match=Match(keyword="field_value_42")),
    ]
).json()
```

The trick is to move more specific `HasIdCondition` before `Filter` (which is the least specific of the three types).

Output:

```json
{
  "must": [
    {
      "has_id": [
        42,
        43
      ]
    },
    {
      "geo_bounding_box": null,
      "geo_radius": null,
      "key": "field_name",
      "match": {
        "integer": null,
        "keyword": "field_value_42"
      },
      "range": null
    }
  ],
  "must_not": null,
  "should": null
}
```

**Comment by generall** [comment]
Thanks for reporting this! This part of code is actually auto-generated, I will see how can I patch it

**Comment by generall** [comment]
Fixed in v0.3.12
